psl-cli/src/test/java/org/linqs/psl/cli/CLITest.java

```diff
@@ -36,18 +41,25 @@
     public static final String RESOURCES_BASE_FILE = ".resources";
     public static final String BASE_DATA_DIR_NAME = "data";
     public static final String BASE_MODELS_DIR_NAME = "models";
+    public static final String BASE_ONLINE_ACTIONS_DIR_NAME = "onlineActions";
+    public static final String SERVER_TEMP_FILE_NAME_PREFIX = "onlinePSLServer";
+    public static final String SERVER_TEMP_FILE_NAME_SUFFIX = ".tmp";
 
     protected final String outDir;
     protected final String resourceDir;
     protected final String baseDataDir;
     protected final String baseModelsDir;
+    protected final String baseOnlineActionsDir;
+    protected final String serverTempFileDir;
 
     public CLITest() {
         outDir = Paths.get(System.getProperty("java.io.tmpdir"), PREFIX + "_" + this.getClass().getName()).toString();
 
         resourceDir = (new File(this.getClass().getClassLoader().getResource(RESOURCES_BASE_FILE).getFile())).getParentFile().getAbsolutePath();
         baseDataDir = Paths.get(resourceDir, BASE_DATA_DIR_NAME).toString();
         baseModelsDir = Paths.get(resourceDir, BASE_MODELS_DIR_NAME).toString();
+        baseOnlineActionsDir = Paths.get(resourceDir, BASE_ONLINE_ACTIONS_DIR_NAME).toString();
+        serverTempFileDir= Paths.get(System.getProperty("java.io.tmpdir")).toString();
```

Comment:
Your temp dirs need to be safe on a multi-tenant machines (like servers).
Use `org.linqs.psl.util.SystemUtils.getTempDir()`.